### PR TITLE
Add a forward pass nn model with dynamism test.

### PR DESCRIPTION
**Pull request by vanbasten23**

### CODE REVIEW

**Comment by JackCaoG** [comment]
you might want to fix the branch

File: test/dynamic_shape_model_test2.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+num_features = 2
+x_train, y_train = make_blobs(n_samples=40, n_features=num_features, cluster_std=1.5, shuffle=True)
+x_train = torch.Tensor(x_train)
+y_train = torch.Tensor(y_train)
+
+# CREATE FAKE TEST DATA
+num_test_samples = 5
+x_test = torch.ones_like(torch.empty(num_test_samples, num_features))
+x_test[0][0] = 0
+y_test = torch.ones_like(torch.empty(num_test_samples*2-1))
```
[the remark this answers is not in the document]
Comment (miladm) [reply test/dynamic_shape_model_test2.py:35]: Shouldn't it be:

```
y_test = torch.ones_like(torch.empty(num_test_samples*2))
y_test[0] = 0
y_test = y_test.to(dev)
y_test = torch.nonzero(y_test)
```

The above sets `y_test` upper bound to `20` and it's dynamic bound to `19` which is consistent with the characteristics of `x_test`. Wdyt @vanbasten23?

[the remark this answers is not in the document]
Comment (vanbasten23) [reply test/dynamic_shape_model_test2.py:35]: > did you verify this implementation works? since `y_test` is not dynamic, I would imaging line 51 fails. No?

Yes it would fail at line 51.

[the remark this answers is not in the document]
Comment (vanbasten23) [reply test/dynamic_shape_model_test2.py:35]: > Shouldn't it be:
> 
> ```
> y_test = torch.ones_like(torch.empty(num_test_samples*2))
> y_test[0] = 0
> y_test = y_test.to(dev)
> y_test = torch.nonzero(y_test)
> ```
> 
> The above sets `y_test` upper bound to `20` and it's dynamic bound to `19` which is consistent with the characteristics of `x_test`. Wdyt @vanbasten23?

That looks good. I implemented this way and it doesn't fail anymore. As a next step, shall we enable the backward pass and check if it works? Also, since our goal is to run on a mask rcnn model, do you want to create a mask rcnn model and directly test it?

**Comment by vanbasten23** [comment]
> you might want to fix the branch

oh I didn't mean to merge this PR at the moment. It's only serves as an experiment script.

File: test/dynamic_shape_model_test2.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vanbasten23) [reply test/dynamic_shape_model_test2.py]: Fixed the test.

**Comment by vanbasten23** [comment]
this is stale.
